website/docs/cli/commands/test.mdx

```diff
@@ -1,13 +1,45 @@
 ---
 page_title: 'Command: test'
-description: Part of the ongoing design research for module integration testing.
+description: >-
+  The `terraform test` command loads and executes Terraform testing files.
 ---
 
 # Command: test
 
-The `terraform test` command is currently serving as part of
-[the module integration testing experiment](/terraform/language/modules/testing-experiment).
+The `terraform test` command reads in Terraform testing files and executes the tests detailed within.
 
-It's not ready for routine use, but if you'd be interested in trying the
-prototype functionality then we'd love to hear your feedback. See the
-experiment details page linked above for more information.
+The `test` command, and the test file syntax, are targeted at module authors wishing to validate and test their shared modules. It is also possible to use the `test` command to validate root modules.
+
+## Usage
+
+Usage: `terraform test [options]`
+
+This command searches the current directory and the specified testing directory (`tests`, by default) for any Terraform testing files, and executes the tests as specified. Test files and their syntax are discussed in detail within the [Tests](/terraform/language/tests) language page.
+
+Terraform will then execute a series of Terraform plan or apply commands, according to the specifications within the test files, and validate the relevant plan and state files, again, according to the specifications within the test files.
+
+> **Warning:** The Terraform test command can create real infrastructure than can cost you money. Read the [Terraform Test Cleanup](#terraform-test-cleanup) section for best practices on ensuring created infrastructure is destroyed.
+
+## General Options
+
+The following options apply to the Terraform `test` command:
+
+* `-filter=testfile` Limits the `terraform test` operation only to the specified test files.
+
+* `-json` Displays machine-readable JSON output for the testing results.
+
+* `-test-directory=<relative directory>` Override the directory that Terraform will look into for testing files. Note, Terraform always loads testing files within the main configuration directory. The default testing directory is `tests`.
+
+* `-verbose` Prints out the state for each `command=apply` `run` block after execution, or the plan for each `command=plan` `run` block.
```

Comment:
Can we use some other syntax for `command=apply` `run`? It looks weird to have the two code blocked things separated by a space in the same line.

Reply:
What about something like this?

```
* `-verbose` - Prints out the plan or state for each `run` block within a test file, based on the `command` attribute of each `run` block.
```

Reply:
Love that! Thank you for suggesting ✨